native/src/modules/categories/components/CategoryListContent.js

```diff
@@ -0,0 +1,179 @@
+    const { language, navigation, navigateToInternalLink } = this.props
+
+    console.log(url)
+    if (url.includes('.pdf')) {
+      navigation.navigate('PDFViewModal', { url })
+    } else if (url.includes('.png') || url.includes('.jpg')) {
+      navigation.navigate('ImageViewModal', { url })
+    } else if (navigateToInternalLink && HIJACK.test(url)) {
+      navigateToInternalLink({
+        url,
+        language
+      })
+    } else {
+      Linking.openURL(url).catch(err => console.error('An error occurred', err))
+    }
+  }
+
+  alterResources = (node: HTMLNode) => {
+    const { cacheDictionary } = this.props
+    if (node.attribs && cacheDictionary) {
+      if (node.attribs.href) {
+        console.debug(`Found href: ${decodeURI(node.attribs.href)}`)
+        const newResource = cacheDictionary[decodeURI(node.attribs.href)]
+        if (newResource) {
+          console.debug(`Replaced ${node.attribs.href} with ${newResource}`)
+          node.attribs = { ...(node.attribs || {}), href: newResource }
+          return node
+        }
+      } else if (node.attribs.src) {
+        console.debug(`Found src: ${decodeURI(node.attribs.src)}`)
+        const newResource = cacheDictionary[decodeURI(node.attribs.src)]
+        if (newResource) {
+          console.debug(`Replaced ${node.attribs.src} with ${newResource}`)
+          node.attribs = { ...(node.attribs || {}), src: newResource }
+          return node
+        }
+      }
+    }
+  }
+
+  // see https://github.com/archriss/react-native-render-html/issues/286
+  renderBulletLists: RendererFunction = (htmlAttribs, children, convertedCSSStyles, passProps) => {
+    const { language } = this.props
+    const style = _constructStyles({
+      tagName: 'ul',
+      htmlAttribs,
+      passProps,
+      styleSet: 'VIEW'
+    })
+    const { allowFontScaling, rawChildren, nodeIndex, key, baseFontStyle, listsPrefixesRenderers } = passProps
+    const baseFontSize = baseFontStyle.fontSize
+    children = children && children.map((child, index) => {
+      const rawChild = rawChildren[index]
+      let prefix = false
+      const rendererArgs = [
+        htmlAttribs,
+        children,
+        convertedCSSStyles,
+        {
+          ...passProps,
+          index
+        }
+      ]
+      const rtlMargin = 10
+      const bulletSizeRelativeToFont = 2.8
+      if (rawChild) {
+        if (rawChild.parentTag === 'ul' && rawChild.tagName === 'li') {
+          prefix = listsPrefixesRenderers && listsPrefixesRenderers.ul ? listsPrefixesRenderers.ul(...rendererArgs) : (
+            <View style={{
+              marginRight: 10,
+              width: baseFontSize / bulletSizeRelativeToFont,
+              height: baseFontSize / bulletSizeRelativeToFont,
+              marginTop: baseFontSize / 2,
+              borderRadius: baseFontSize / bulletSizeRelativeToFont,
+              backgroundColor: 'black',
+              marginLeft: RTL_LANGUAGES.includes(language) ? rtlMargin : 0
```

Comment:
Why do you set `marginRight` to 10 always but not `marginLeft`? You should probably use the var for `marginRight` as well.

Reply:
Actually most of the code is more or less copy/pasted from the menioned [github issue](https://github.com/archriss/react-native-render-html/issues/286). I tried renaming the arguments a little hopefully it is more clear now.